src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -556,6 +568,10 @@ bool emberAfOperationalCredentialsClusterAddNOCCallback(app::CommandHandler * co
     // We might have a new operational identity, so we should start advertising it right away.
     app::DnssdServer::Instance().AdvertiseOperational();
 
+    // The Fabric Index associated with the armed fail-safe context SHALL be updated to match the Fabric
+    // Index just allocated.
+    failSafeContext.SetNocCommandInvoked(fabricIndex);
```

Comment:
As  followup: We should roll back the fabric addition if the ACL failed to be added properly....

Reply:
https://github.com/project-chip/connectedhomeip/issues/15491